### PR TITLE
Rename target_dataframe_name parameter to target_dataframe_index

### LINKED ISSUE #348: Rename target_dataframe_name parameter to target_dataframe_index
Compose previously used the terminology `target_entity` to refer to the target column. When Featuretools updated to change `target_entity` to `target_dataframe_name`, a corresponding change was made in Compose in PR #241.

I don't believe this change should have been made. In Compose the `target_entity` actually refers to the name of a column in a dataframe, not the name of a `ft.Entity` object. The current parameter name is confusing and should be changed to be more clear.

Suggestion: rename the `target_dataframe_name` parameter to `target_dataframe_index`.

### PR DESCRIPTION
**Pull request by gsheni**
- Closes #348

### CODE REVIEW

File: composeml/label_times/description.py [hunk 1]
```diff
@@ -20,11 +20,9 @@ def describe_label_times(label_times):
     metadata = label_times.settings
     target_column = metadata["label_times"]["target_columns"][0]
     target_type = metadata["label_times"]["target_types"][target_column]
-    target_dataframe_name = metadata["label_times"]["target_dataframe_name"]
```
Comment (gsheni) [inline composeml/label_times/description.py:23]: I am unsure about what this does and if we should remove this or not.

File: docs/source/release_notes.rst [hunk 2]
```diff
@@ -8,13 +8,18 @@ Future Release
         * Update to avoid error with a categorical target with unused categories (:pr:`349`)
     * Changes
         * Transition to pure pyproject.toml for project metadata (:pr:`351`)
```
Comment (thehomebrewnerd) [inline docs/source/release_notes.rst:10]: I just happened to notice that we still have a `setup.py` hanging around from PR 351. We could delete it here or do it in a separate PR.

File: composeml/label_times/description.py (hunk 1 above)
Comment (thehomebrewnerd) [reply composeml/label_times/description.py:23]: When I look at the `LabelTimes` class definition, it seems like we have some pretty confusing terminology. I think in one sense `target_column_name` refers to the column we want to use as the "target" for creating labels (as in, I want to create labels for my customers using `customer_id`).  

Then `target_column` here seems to refer to the column that is the machine learning target column (the label column, as in, `total_spent` for each customer).

I'm not sure how best to resolve this off the top of my head, but we should probably only use target in one case.

File: docs/source/release_notes.rst (hunk 2 above)
Comment (gsheni) [reply docs/source/release_notes.rst:10]: Removed

File: composeml/label_times/description.py (hunk 1 above)
Comment (gsheni) [reply composeml/label_times/description.py:23]: @jeff-hernandez Any idea what to do here?

Comment (jeff-hernandez) [reply composeml/label_times/description.py:23]: I think the intent is to use the same terminology as featuretools. For example, in the past, we used `target_entity` to specify which entity to create features for. Similarly, the idea was the use `target_entity` to specify which entity to create labels for. The reason we put it in the metadata is so that we have the settings for reconstructing the label times.

Comment (jeff-hernandez) [reply composeml/label_times/description.py:23]: Maybe using `target_column_index` instead of `target_column_name` can help clarify

File: composeml/label_maker.py [hunk 3]
```diff
@@ -14,23 +14,23 @@ class LabelMaker:
...
+        target_dataframe_index,
         time_index,
         labeling_function=None,
         window_size=None,
     ):
         """Creates an instance of label maker.
 
         Args:
-            target_dataframe_name (str): Dataframe on which to make labels.
+            target_dataframe_index (str): Column on which to make labels.
```
Comment (jeff-hernandez) [inline composeml/label_maker.py:25]: `Column` -> `Index of the target dataframe`?

File: composeml/label_times/description.py [hunk 4]
```diff
@@ -20,11 +20,9 @@ def describe_label_times(label_times):
...
 
     settings = {
         "target_column": target_column,
-        "target_dataframe_name": target_dataframe_name,
```
Comment (jeff-hernandez) [inline composeml/label_times/description.py:27]: I think we may still want to track `target_dataframe_index` in the settings

**Review by jeff-hernandez — APPROVED** [review]
Overall looks good, just had a couple of comments.

File: composeml/label_maker.py (hunk 3 above)
Comment (gsheni) [reply composeml/label_maker.py]: Fixed: https://github.com/alteryx/compose/pull/353/commits/2f817c80ead4acefbc6b8cd886c7916e84170e84

File: composeml/label_times/description.py (hunk 4 above)
Comment (gsheni) [reply composeml/label_times/description.py:27]: Fixed https://github.com/alteryx/compose/pull/353/commits/ae874cbd2c5eae062b007fb8121e8dc9a792b2b9

**Review by jeff-hernandez — APPROVED** [review]
